src/AppInstallerCLICore/Workflows/WorkflowBase.cpp

```diff
@@ -432,6 +432,15 @@ namespace AppInstaller::CLI::Workflow
         }
     }
 
+    void CheckForMultiQuery(Execution::Context& context)
+    {
+        if (context.Args.GetCount(Execution::Args::Type::MultiQuery) == 1)
+        {
+            context.Args.AddArg(Execution::Args::Type::Query, context.Args.GetArg(Execution::Args::Type::MultiQuery));
+            context.Args.RemoveMultiQueryArg();
+        }

```

Comment:
Do you think the whole logic should be in ContextArg class to ensure  args integrity that whenever we remove MultiQuery we add the Query, and do that in arg parsing phase?
Or you think ContextArg should be simple and not handle  complex logic?

I personally think handling it in arg parsing phase may be easier in workflow and less error prone.

Reply:
I hadn't even thought about doing it in the arg parsing phase instead of the execution. That does make it a lot cleaner.

I'm moving the logic to a single method in ExecutionArgs, and calling it at the end of `Command::ParseArguments()`. It ends up being simple and in a single place, it also helps with the arg validation because I can set the categories for MultiQuery assuming that it always has more than one value.